core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputReader.java

```diff
@@ -489,20 +515,25 @@ public ExcelInputReaderConfig() {
       super("ExcelInputReader");
     }
 
-    public void validate() {
+    public void validate(FailureCollector collector) {
       if (!containsMacro("sheetValue") && sheet.equalsIgnoreCase(SHEET_NO) && !StringUtils.isNumeric(sheetValue)) {
-        throw new IllegalArgumentException("Invalid sheet number. The value should be greater than or equals to zero.");
+        collector.addFailure(
+            String.format("Invalid sheet number: '%s'.", sheetValue),
+            "The value should be greater than or equal to zero.")
+            .withConfigProperty(SHEET_VALUE);
       }
 
       if (!(Strings.isNullOrEmpty(tableExpiryPeriod)) && (Strings.isNullOrEmpty(memoryTableName))) {
-        throw new IllegalArgumentException("Value for Table Expiry period is valid only when file tracking table " +
-                                             "is specified. Please specify file tracking table name.");
+        collector.addFailure(
+            "Value for Table Expiry Period is valid only when file tracking table is specified.", null)
+            .withConfigProperty(TABLE_EXPIRY_PERIOD);
       }
 
       if (!Strings.isNullOrEmpty(rowsLimit) && !StringUtils.isNumeric(rowsLimit)) {
-        throw new IllegalArgumentException(String.format("Invalid row limit: %s. " +
-                                                           "Numeric value expected.", rowsLimit));
+        collector.addFailure(String.format("Invalid row limit: '%s'.", rowsLimit), "Numeric value expected.")
+            .withConfigProperty(ROWS_LIMIT);
       }
+      collector.getOrThrowException();
```

Comment:
should not be here, the caller should decide whether to call it. In the case of configurePipeline(), it looks like more things can be validated.

Reply:
Fixed